Dockerfile

```diff
@@ -56,6 +56,13 @@ WORKDIR /opt/guacamole
 # Copy artifacts from builder image into this image
 COPY --from=builder /opt/guacamole/ .
 
+# Turn on the Remote IP Valve
+RUN sed -i 's+^\(\( *\)</Host>\)+\2  <Valve className="org.apache.catalina.valves.RemoteIpValve" />\n\n\1+' /usr/local/tomcat/conf/server.xml
```

Comment:
Why does the regex care about spaces before the closing tag? And, if it does, shouldn't it use a generic whitespace pattern rather than a space?

What is the impact of configuring the Remote IP Valve with default options? I have only ever seen it with proxy filtering enabled.

Reply:
Thank you for your review @manolan1 👍

> Why does the regex care about spaces before the closing tag?

To be sure, in addition to the `^` anchor, to select the real closing tag, and not for example a comment.

> And, if it does, shouldn't it use a generic whitespace pattern rather than a space?

I used space because the .xml file uses spaces, but I've switched to `\s`, even stronger. Thx.

Reply:
> What is the impact of configuring the Remote IP Valve with default options? I have only ever seen it with proxy filtering enabled.

If you use (as you should) your Docker instance behind a (Apache) reverse proxy :
- without the Remote IP Valve, you'll get the Docker interface IP in the Guacamole logs ;
- with the Remote IP Valve, you'll get the correct client (public) IP in the Guacamole logs.

If you use your Docker instance directly (strange as you won't provide SSL support etc...) :
- with or without the Remote IP Valve, you'll get the correct client (public) IP in the Guacamole logs. The only drawback is that a client with a **local** IP should be able to replace its IP by a fake one in Guacamole logs sending the correct headers (normally sent by the proxy) in his requests (`X-Forwarded-For`).